File: python/541.py
class Solution:
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        l = len(s)
        start = 0
        ans = ""
        while l > 2 * k:
            temp = s[start:start + k]
            ans = ans + temp[::-1] + s[start+k:start +2 *k]
            start += 2 * k
            l -= 2*k
        if l < k:
            temp = s[start:]
            ans = ans + temp[::-1]
        else:
            temp = s[start:start + k]
            ans = ans + temp[::-1] + s[start+k:]
        return ans

File: go/541.go
func reverseStr(s string, k int) string {
	sbyte := []byte(s)
	i := 0
	for ; i < len(s); i += 2 * k {
		j := i
		l := i + k - 1
		if l > len(s)-1 {
			l = len(s) - 1
		}
		swap(sbyte, j, l)
	}
	return string(sbyte)

}

func swap(sb []byte, i, j int) {
	for i < j {
		k := sb[i]
		sb[i] = sb[j]
		sb[j] = k
		i++
		j--
	}
}